Kudu.Core/LinuxConsumption/MeshPersistentFileSystem.cs

```diff
@@ -0,0 +1,160 @@
+        }
+
+        private bool IsKuduShareMounted()
+        {
+            return _fileShareMounted;
+        }
+
+        private void UpdateStatus(bool status, string message)
+        {
+            _fileShareMounted = status;
+            _fileShareMountMessage = message;
+        }
+
+        /// <summary>
+        /// Mounts file share
+        /// </summary>
+        /// <returns></returns>
+        public async Task<bool> MountFileShare()
+        {
+            var siteName = ServerConfiguration.GetApplicationName();
+
+            if (IsKuduShareMounted())
+            {
+                const string message = "Kudu file share mounted already";
+                UpdateStatus(true, message);
+                KuduEventGenerator.Log(_environment).LogMessage(EventLevel.Warning, siteName, nameof(MountFileShare), message);
+                return true;
+            }
+
+            if (!IsLinuxConsumption())
+            {
+                const string message =
+                    "Mounting kudu file share is only supported on Linux consumption environment";
+                UpdateStatus(false, message);
+                KuduEventGenerator.Log(_environment).LogMessage(EventLevel.Warning, siteName, nameof(MountFileShare), message);
+                return false;
+            }
+
+            if (!IsPersistentStorageEnabled())
+            {
+                const string message = "Kudu file share was not mounted since persistent storage is disabled";
+                UpdateStatus(false, message);
+                KuduEventGenerator.Log(_environment).LogMessage(EventLevel.Warning, siteName, nameof(MountFileShare), message);
+                return false;
+            }
+
+            if (!TryGetStorageConnectionString(out var connectionString))
+            {
+                var message = $"Kudu file share was not mounted since {Constants.AzureWebJobsStorage} is empty";
+                UpdateStatus(false, message);
+                KuduEventGenerator.Log(_environment).LogMessage(EventLevel.Warning, siteName, nameof(MountFileShare), message);
+                return false;
+            }
+
+            var errorMessage = await MountKuduFileShare(siteName, connectionString);
+            var mountResult = string.IsNullOrEmpty(errorMessage);
+
+            UpdateStatus(mountResult, errorMessage);
+            KuduEventGenerator.Log(_environment).LogMessage(EventLevel.Informational, siteName,
+                $"Mounting Kudu file share result: {mountResult}", string.Empty);
+
+            return mountResult;
+        }
+
+        public bool GetStatus(out string message)
+        {
+            message = _fileShareMountMessage;
+            return _fileShareMounted;
+        }
+
+        public string GetDeploymentsPath()
+        {
+            if (_fileShareMounted)
+            {
+                return Path.Combine(Constants.KuduFileShareMountPath, "deployments");
```

Comment:
Nit, can replace to **DeploymentCachePath**.

Reply:
Thanks. Will fix this..